packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -221,9 +241,69 @@ class FileDownloader(
     }
   }
 
-  private fun isPatchResponse(responseBody: ResponseBody): Boolean {
-    val mediaType = responseBody.contentType()
-    return mediaType?.let { it.type == "application" && it.subtype == "vnd.bsdiff" } ?: false
+  private data class PatchResponseMetadata(
+    val hasImBsdiff: Boolean,
+    val hasLegacyContentType: Boolean,
+    val statusCode: Int,
+    val deltaBase: String?,
+    val eTag: String?,
+    val expoBaseUpdateId: String?
+  )
+
+  private fun parsePatchResponseMetadata(response: Response): PatchResponseMetadata? {
+    val mediaType = response.body?.contentType()
+    val hasLegacyContentType = mediaType?.let { it.type == "application" && it.subtype == "vnd.bsdiff" } ?: false
```

Comment:
I think we can remove legacy content type logic? EAS no longer uses it as far as I know